finetune/target_models/logistic_regression.py

```diff
@@ -0,0 +1,108 @@
+import spacy
+import numpy as np
+import joblib as jl
+
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.linear_model import LogisticRegression
+from finetune.base import BaseModel
+from finetune.errors import FinetuneError
+
+class GloveEncoder():
+    nlp = spacy.load("en_core_web_lg")
+
+    def encode(self, text, train):
+        batch_glove = np.squeeze(np.asarray([self.nlp(example).vector for example in text]))
+        return batch_glove
+
+class TfidfEncoder():
+
+    def __init__(self):
+        self.encoder = TfidfVectorizer(ngram_range=(1,3), max_features=5000)
+
+    def encode(self, text, train):
+        if train: self.encoder.fit(text)
+        batch_tfidf = np.squeeze(np.asarray(self.encoder.transform(text).toarray()))
+        return batch_tfidf
+
+
+class LogisticRegressionClassifier():
+    """ 
+    Classifies a single document into 1 of N categories using logistic regression.
+
+    :param config: A :py:class:`finetune.config.Settings` object or None (for default config).
+    :param \**kwargs: key-value pairs of config items to override.
+    """
+    defaults = {'encoder': 'glove'}
+    
+
+    def __init__(self, **kwargs):
+        self.LR = LogisticRegression(penalty="l2", max_iter=500, solver="lbfgs", multi_class='multinomial')
+        c = kwargs.pop('C', None)
+        if c: self.LR.C = c
```

Comment:
So many if statement one liners.

I'm convinced this was intended to be a personal attack.

Reply:
if madison_is_angry: keep_doing_it()